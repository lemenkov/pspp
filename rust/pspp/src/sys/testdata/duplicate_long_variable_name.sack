# File header.
"$FL2"; s60 "$(#) SPSS DATA FILE PSPP synthetic test file";
2; 4; 1; 0; -1; 100.0; "01 Jan 11"; "20:53:52"; s64 ""; i8 0 *3;

# Numeric variables.
2; 0; 0; 0; 0x050800 *2; s8 "LONGVARI";
2; 0; 0; 0; 0x050800 *2; s8 "LONGVA_A";
2; 0; 0; 0; 0x050800 *2; s8 "LONGVA_B";
2; 0; 0; 0; 0x050800 *2; s8 "LONGVA_C";

# Long variable names.
7; 13; 1; COUNT (
"LONGVARI=_Invalid"; i8 9;
"LONGVARI=$Invalid"; i8 9;
"LONGVARI=#Invalid"; i8 9;
"LONGVA_A=LongVariableName"; i8 9;
"LONGVA_B=LONGVARIABLENAME"; i8 9;
);

# Character encoding record.
7; 20; 1; 12; "windows-1252";

# Dictionary termination record.
999; 0;
