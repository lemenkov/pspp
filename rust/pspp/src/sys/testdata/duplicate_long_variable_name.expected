Warning at file offsets 0x140 to 0x1aa: In long variable name record: Invalid long name.  "_Invalid" may not be used as an identifier because it begins with disallowed character '_'.

Warning at file offsets 0x140 to 0x1aa: In long variable name record: Invalid long name.  "$Invalid" may not be used as an identifier because it begins with disallowed character '$'.

Warning at file offsets 0x140 to 0x1aa: In long variable name record: Invalid long name.  "#Invalid" may not be used as an identifier because it begins with disallowed character '#'.

Duplicate long variable name LONGVARIABLENAME.

╭──────────────────────┬────────────────────────╮
│       Created        │    01-JAN-2011 20:53:52│
├──────────────────────┼────────────────────────┤
│Writer Product        │PSPP synthetic test file│
├──────────────────────┼────────────────────────┤
│       Compression    │SAV                     │
│       Number of Cases│Unknown                 │
╰──────────────────────┴────────────────────────╯

╭─────────┬─╮
│Variables│4│
╰─────────┴─╯

╭────────────────┬────────┬─────┬─────────────────┬─────┬─────┬─────────┬────────────┬────────────┬──────────────╮
│                │Position│Label│Measurement Level│ Role│Width│Alignment│Print Format│Write Format│Missing Values│
├────────────────┼────────┼─────┼─────────────────┼─────┼─────┼─────────┼────────────┼────────────┼──────────────┤
│LONGVARI        │       1│     │                 │Input│    8│Right    │F8.0        │F8.0        │              │
│LongVariableName│       2│     │                 │Input│    8│Right    │F8.0        │F8.0        │              │
│LONGVA_B        │       3│     │                 │Input│    8│Right    │F8.0        │F8.0        │              │
│LONGVA_C        │       4│     │                 │Input│    8│Right    │F8.0        │F8.0        │              │
╰────────────────┴────────┴─────┴─────────────────┴─────┴─────┴─────────┴────────────┴────────────┴──────────────╯
